File: 15-18/CrashUtils-master/CrashUtils/app/src/main/java/com/wx/crashdemo/MainActivity.java
package com.wx.crashdemo;

import android.Manifest;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Toast;

import com.orhanobut.logger.Logger;
import com.tbruyelle.rxpermissions2.RxPermissions;
import com.wx.crashlibrary.manager.CrashManager;
import com.wx.crashlibrary.manager.DeviceInfoManager;

import java.util.HashMap;
import java.util.Map;

import io.reactivex.functions.Consumer;

public class MainActivity extends AppCompatActivity implements View.OnClickListener{

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        RxPermissions rxPermissions = new RxPermissions(this);
        rxPermissions.request(new String[]{Manifest.permission.READ_PHONE_STATE,Manifest.permission.WRITE_EXTERNAL_STORAGE})
                .subscribe(new Consumer<Boolean>() {
                    @Override
                    public void accept(Boolean aBoolean) throws Exception {
                        if (!aBoolean) {
                            Toast.makeText(MainActivity.this,"无法测试，请授权",Toast.LENGTH_SHORT).show();
                            finish();
                        }
                    }
                });

        findViewById(R.id.btn_1).setOnClickListener(this);
        findViewById(R.id.btn_2).setOnClickListener(this);
        findViewById(R.id.btn_3).setOnClickListener(this);
        findViewById(R.id.btn_4).setOnClickListener(this);
        findViewById(R.id.btn_5).setOnClickListener(this);
        findViewById(R.id.btn_6).setOnClickListener(this);
        findViewById(R.id.btn_8).setOnClickListener(this);
    }

    @Override
    public void onClick(View view) {
        switch (view.getId()) {
            case R.id.btn_1:
                int a = 1/0;
                break;
            case R.id.btn_2:
                String b = null;
                b.equals("");
                break;
            case R.id.btn_3:
                Map<Integer,Bitmap> bitmapMap = new HashMap<>();
                for (int i = 0; i <100 ;i++) {
                    bitmapMap.put(i, BitmapFactory.decodeResource(getResources(),R.mipmap.icon_2k));
                }
                break;
            case R.id.btn_4:
                try {
                    int c = 1/0;
                } catch (Exception e) {

                }
                break;
            case R.id.btn_5:
                Logger.json(DeviceInfoManager.builder(MainActivity.this).getJsonData());
                break;
            case R.id.btn_6:
                new Thread(){
                    @Override
                    public void run() {
                        int d= 1/0;
                    }
                }.start();
                break;
            case R.id.btn_8:
                CrashManager.getInstance().init().clearDefaultLog();
                break;
        }
    }

}

File: Android-Demo/LBSMap/GaodeMaster/app/src/main/java/com/wx/gaodemaster/MainActivity.kt
package com.wx.gaodemaster

import android.Manifest
import android.databinding.ViewDataBinding
import android.view.View
import com.tbruyelle.rxpermissions2.RxPermissions
import com.wx.gaodemaster.actvity.MapActivity
import com.wx.gaodemaster.base.BaseActivity
import com.wx.gaodemaster.databinding.ActivityMainBinding
import io.reactivex.functions.Consumer

class MainActivity : BaseActivity() {

    lateinit var binding: ActivityMainBinding

    override fun getLayouId(): Int {
        return R.layout.activity_main
    }

    override fun setDataBinding(binding: ViewDataBinding) {
        this.binding = binding as ActivityMainBinding;
    }

    override fun initData() {
        initPermissions()
        binding.btnMap.setOnClickListener(View.OnClickListener {
            openActivity(MapActivity::class.java)
        })
    }

    fun initPermissions() {
        val rxPermission: RxPermissions = RxPermissions(this)
        rxPermission.requestEach(Manifest.permission.ACCESS_COARSE_LOCATION,
                Manifest.permission.ACCESS_FINE_LOCATION,
                Manifest.permission.WRITE_EXTERNAL_STORAGE,
                Manifest.permission.READ_EXTERNAL_STORAGE,
                Manifest.permission.READ_PHONE_STATE)
                .subscribe(Consumer {
                    if (it.granted) {   // 用户已经同意该权限

                    } else if (it.shouldShowRequestPermissionRationale) {
                        // 用户拒绝了该权限，没有选中『不再询问』（Never ask again）,那么下次再次启动时，还会提示请求权限的对话框
                        showToast("请授予权限")
                        finish()
                    } else {
                        // 用户拒绝了该权限，并且选中『不再询问』
                        showToast("请在权限界面授予权限")
                        finish()
                    }
                })

    }

}
